From bfebab6f37f7f9fce164aa6896f2053d59229ed7 Mon Sep 17 00:00:00 2001
Message-ID: <bfebab6f37f7f9fce164aa6896f2053d59229ed7.1719691458.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1719691458.git.sharlatanus@gmail.com>
References: <cover.1719691458.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Mon, 24 Jun 2024 21:21:13 +0100
Subject: [PATCH 20/26] gnu: Add go-atomicgo-dev-cursor.

* gnu/packages/golang-xyz.scm (go-atomicgo-dev-cursor): New variable.

Change-Id: I7d742bf6056b3169999157af55cc1c002e6aceb9
---
 gnu/packages/golang-xyz.scm | 26 ++++++++++++++++++++++++++
 1 file changed, 26 insertions(+)

diff --git a/gnu/packages/golang-xyz.scm b/gnu/packages/golang-xyz.scm
index 0f7d462816..4bdcc05c47 100644
--- a/gnu/packages/golang-xyz.scm
+++ b/gnu/packages/golang-xyz.scm
@@ -85,6 +85,32 @@ (define-module (gnu packages golang-xyz)
 ;;; Libraries:
 ;;;
 
+(define-public go-atomicgo-dev-cursor
+  (package
+    (name "go-atomicgo-dev-cursor")
+    (version "0.2.0")
+    (source
+     (origin
+       (method git-fetch)
+       (uri (git-reference
+             (url "https://github.com/atomicgo/cursor")
+             (commit (string-append "v" version))))
+       (file-name (git-file-name name version))
+       (sha256
+        (base32 "0ds85nyd3dnjr961x9g5kflx1qdb92vn7n6wc4jbk0fjjzbrnh5s"))))
+    (build-system go-build-system)
+    (arguments
+     (list
+      #:import-path "atomicgo.dev/cursor"))
+    (home-page "https://atomicgo.dev/cursor")
+    (synopsis "Moving terminal cursor in Golang")
+    (description
+     "Package cursor contains cross-platform methods to move the terminal cursor in
+different directions.  This package can be used to create interactive CLI tools
+and games, live charts, algorithm visualizations and other updatable output of
+any kind.")
+    (license license:expat)))
+
 (define-public go-bazil-org-fuse
   (package
     (name "go-bazil-org-fuse")
-- 
2.41.0

